### PR TITLE
CLDR-14395 API for Auth endpoints

### PR DESCRIPTION
**Pull request by srl295**
CLDR-14395

- API for login/logout/info

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                    schema = @Schema(implementation = LoginResponse.class))),
+            @APIResponse(
+                responseCode = "401",
+                description = "Authorization required"),
+            @APIResponse(
+                responseCode = "403",
+                description = "Forbidden, account may be locked"),
+        })
+    public Response login(
+        @Context HttpServletRequest hreq,
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java:54]: long function, could use more subroutines

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+        try {
+            LoginResponse resp = new LoginResponse();
+
+            // we start with the user
+            User user = CookieSession.sm.reg.get(request.password, request.email, ip, false);
+            if(user == null) {
+                return Response.status(403, "Login failed").build();
+            } if(UserRegistry.userIsLocked(user)) {
+                return Response.status(403, "User Locked").build();
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java:71]: formatting

**Review by btangmu — APPROVED** [review]
great modernization and encapsulation!

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java (hunk 2 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java:71]: need to find out how to run the eclipse formatter from the cli

Comment (btangmu) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java:71]: for js I'm happy with npx prettier -- there is a java plugin, but I haven't tried it: https://github.com/jhipster/prettier-java

btw what are you using on vue files?

**Comment by srl295** [comment]
`/info` and `/login` return the same results for simplicity. Reproduced here.

```json
{
  "newlyLoggedIn": false,
  "sessionId": "1Yfj5oDfGEi9uT3rY7ardNM1D3U",
  "user": {
    "email": "tevaalinguist.zb4yfawzc@fjqk.cherokee.example.com",
    "emailHash": "31ddce0201685ea79e5dba5232b2d7a7",
    "id": 2475,
    "level": "tc",
    "locales": "",
    "name": "TevaALinguist_TESTER_",
    "org": "Cherokee",
    "organization": "cherokee",
    "permissions": {
      "userIsLocked": false,
      "userCanImportOldVotes": true,
      "userCanMonitorForum": true,
      "userIsVetter": false,
      "userIsAdmin": false,
      "userIsTC": true,
      "userCanUseVettingSummary": true
    },
    "permissionsJson": {},
    "userlevel": 1,
    "voteCount": 20,
    "voteCountMenu": [
      4,
      20,
      1000
    ],
    "voterInfo": {
      "level": "tc",
      "locales": [],
      "name": "TevaALinguist_TESTER_",
      "organization": "cherokee"
    }
  }
}
```

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java (hunk 2 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java]: cleaned up

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java:46]: refactored a bit. Lot of tech debt in the user management …

**Review by btangmu — APPROVED** [review]

**Pull request merged**